package-res/resources/web/vizapi/VizController.js

```diff
@@ -717,28 +717,28 @@ pentaho.VizController.getRgbGradientFromMultiColorHex = function(value, min, max
   var range = max-min;
 
   if(range <= 0) {
-    var start = colors.length-1;
-    var end = start;
+    var startStep = colors.length-1;
+    var endStep = startStep;
   } else {
-    var start = Math.floor(((value-min)/range) * steps);
-    var end = Math.ceil(((value-min)/range) * steps);
+    var inRange = ((value-min)/range);
+    var startStep = Math.floor(inRange * steps);
+    var endStep = Math.ceil(inRange * steps);
```

Comment:
Scope of variables: the variables `inRange`, `startStep` and `endStep` are used outside of this `if` block, and as such they should be declared outside.